client/src/components/login/RegisterFormExternal.vue

```diff
@@ -0,0 +1,153 @@
+<template>
```

Comment:
Would definitely prefer a refactored version of this that doesn't duplicate all this just to change the order.  In general, what I'd suggest if you needed to break down sections and allow for ordering is pulling the component apart into more, smaller sections -- i.e., here, the contents of the two collapse sections that you can then prioritize in order based on oidc-enabled (or, even better, auth_conf preferences set by admins).

Reply:
So, would that mean I can just incorporate this into the `RegisterForm` component, removing the need for a separate one (`RegisterFormExternal`)?

Reply:
Yep.

Reply:
nice! So, it's ok if I make a new component for `OIDCRegister.vue` and `LocalRegistration.vue`?

Reply:
Yeah, feel free to come up with a better name, too -- that's just what my fingers spat out without thinking :)

Reply:
Ok, also, when you say `auth_conf preferences set by admins`, would that mean adding a preferOIDC option to `auth.conf.xml` like:
`<prefer-OIDC>true</prefer-OIDC>`
Thanks!

Reply:
Right; an attribute on the actual `auth` option is what I had in mind (`preferred=true` or the like), but looking at it again that's a mess.  In an ideal world we could unify auth_conf.xml and the various oidc ones into just a single, comprehensive `auth_conf.xml`, which would have this.

Looking at it again, though, a much simpler option is to follow suit with other login customizations like (`show_welcome_with_login`) and add it to the standard galaxy config, like `login_page_display_order` or something.

Reply:
For `auth_conf.xml` the backend already tries the configured authenticators in the order they are listed in the config file, but all these authenticators have a single username/password UI and the authenticators are tried automatically on the basis of (optional) filters (e.g. on the email address domain).

Reply:
@nsoranzo Yeah, just passing back to the client the ordering of entries auth_conf would work great if it had all the options -- but none of the OIDC options will be in there given the current config organization that has them in a separate file altogether.  I guess we could add something as a quick temp fix for that like:

```
<auth>
    <authenticator>
        <type>localdb</type>
        <options>
            <allow-password-change>true</allow-password-change>
        </options>
    </authenticator>

    <authenticator>
       <!-- expand/include OIDC options here, in order -->
        <type>OIDC</type>
</auth>
```
Where you could move around and order OIDC as a placeholder for all the stuff in oidc_config, but I think just merging the oidc_config stuff in for a single auth configfile might be a better solution in the long run.  For this first pass though I'm happy with a (potentially brief-lived) config handle in galaxy.yml.